blockchain-agent-proof/src/main/java/com/sphereon/alfresco/blockchain/agent/proof/tasks/ProofVerifyPendingRegistrationsTask.java

```diff
@@ -0,0 +1,51 @@
+package com.sphereon.alfresco.blockchain.agent.proof.tasks;
+
+import com.sphereon.alfresco.blockchain.agent.proof.ProofApiUtils;
+import com.sphereon.alfresco.blockchain.agent.rest.model.VerifyContentAlfrescoResponse;
+import com.sphereon.alfresco.blockchain.agent.tasks.scheduled.VerifyPendingRegistrationsTask;
+import com.sphereon.alfresco.blockchain.agent.utils.Signer;
+import com.sphereon.libs.authentication.api.TokenRequest;
+import com.sphereon.sdk.blockchain.proof.api.VerificationApi;
+import com.sphereon.sdk.blockchain.proof.model.ContentRequest;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Component;
+
+@Component
+@Profile("sphereon-proof-api")
+public class ProofVerifyPendingRegistrationsTask implements VerifyPendingRegistrationsTask {
+    private final VerificationApi bcProofVerificationApi;
+    private final TokenRequest tokenRequester;
+    private final ProofApiUtils utils;
+    private Signer signer;
+    private final String proofApiConfigName;
+
+    public ProofVerifyPendingRegistrationsTask(final VerificationApi bcProofVerificationApi,
+                                               final TokenRequest tokenRequester,
+                                               final ProofApiUtils utils,
+                                               final Signer signer,
+                                               @Value("${blockchain.config-name:#{null}}") final String proofApiConfigName) {
+        this.bcProofVerificationApi = bcProofVerificationApi;
+        this.tokenRequester = tokenRequester;
+        this.utils = utils;
+        this.signer = signer;
+        this.proofApiConfigName = proofApiConfigName;
+    }
+
+    @Override
+    public VerifyContentAlfrescoResponse verifyHash(byte[] contentHash) {
+        tokenRequester.execute();
+        var contentRequest = new ContentRequest();
+        contentRequest.setContent(contentHash);
+        contentRequest.setHashProvider(ContentRequest.HashProviderEnum.CLIENT);
```

Comment:
I have seen at least 3 times that the content request is being created in the same way. Maybe move that or introduce an abstract super class for the taks

Reply:
Two places right now. Let's leave this for later.